llvm/lib/MC/MCParser/DarwinAsmParser.cpp

```diff
@@ -613,7 +613,7 @@ bool DarwinAsmParser::parseDirectiveLinkerOption(StringRef IDVal, SMLoc) {
       return TokError("expected string in '" + Twine(IDVal) + "' directive");
 
     std::string Data;
-    if (getParser().parseEscapedString(Data))
+    if (getParser().parseEscapedString(Data, false))
```

Comment:
I guess the intent of the warning is to help finding missing quotes in such pieces of assembly:

```asm
.string "asdf
// lots of directives / code here
.string "qwer"
```

The assembler would complain about unknown instruction mnemonic "qwer" because it parsed everything after the very first quote as an argument of the first `.string` directive. This consideration should apply to `.linker_option` directive as well, which is unlikely to expect newlines in the argument. That is, I would prohibit newlines in the argument of most, if not all, directives calling `parseEscapedString`.

Reply:
So your suggestion is to emit `Error()` instead of `Warning()` for all such directives ?

Reply:
I agree with you. Such a code with newlines in strange places most likely can only appear if it is written manually.
So i i think that prohibiting of newlines in all `parseEscapedString()` places is a good idea

Reply:
And if it is just a warning, it will most likely be ignored

Reply:
> No, I mean the warning should be issued for most/all of such directives, not only for .string/.ascii. Sorry if I wasn't clear.

No problem :)
my understanding was very close.
Okay, then i'm going to remove the second argument of `parseEscapedString` and emit warning unconditionally.

Reply:
Then i should extend test cases

Reply:
A couple of different directives should be enough, other directives do not add new context.

Reply:
Done